### PR TITLE
[metricbeat]kubernetes: add persistentvolumeclaim from kube-state-metrics

### PR DESCRIPTION
**Pull request by odacremolbap**
kubernetes persistent volume claim metrics for kube-state-metrics.
refer to:

https://github.com/kubernetes/kube-state-metrics/blob/release-1.8/docs/persistentvolumeclaim-metrics.md

Note: master branch ksm document above adds a metric family that is not supported yet, and won't be until next ksm is released.

related to elastic/beats#14821

### CODE REVIEW

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - name: phase
+      type: keyword
+      description: PVC phase.
+    - name: access_mode
+      type: keyword
+      description: Access mode.
+    - name: storage_class
+      type: keyword
+      description: Storage class for the PVC.
+    - name: labels.*
```
Comment (exekias) [inline metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/fields.yml]: do you know if this labels are exactly Kubernetes labels? If that's the case we could map them under the common `kubernetes.labels`

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		},
+	}, nil
+}
+
+// Fetch prometheus metrics and treats those prefixed by mb.ModuleDataKey as
+// module rooted fields at the event that gets reported
+func (m *persistentvolumeclaimMetricSet) Fetch(reporter mb.ReporterV2) {
+	events, err := m.prometheus.GetProcessedMetrics(m.mapping)
+	if err != nil {
+		m.Logger().Error(err)
```
Comment (sayden) [inline metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim.go]: You can write the signature of the method like `func (m *persistentvolumeclaimMetricSet) Fetch(reporter mb.ReporterV2) error` and `return err` here directly. The under the hood implementation will do the `m.Logger().Error(err)` and the `reporter.Error(err)`.

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/fields.yml (hunk 1 above)
Comment (odacremolbap) [reply metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/fields.yml]: It is.

I moved it to the module location. 
Will need to do so with PV and Services (on a different PR)

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim.go (hunk 2 above)
Comment (odacremolbap) [reply metricbeat/module/kubernetes/state_persistentvolumeclaim/state_persistentvolumeclaim.go]: thanks a lot @sayden
will use that func signature from now on.

File: metricbeat/module/kubernetes/_meta/fields.yml [hunk 3]
```diff
@@ -11,3 +11,8 @@
       description: >
         Information and statistics of pods managed by kubernetes.
       fields:
+        - name: labels.*
```
Comment (exekias) [inline metricbeat/module/kubernetes/_meta/fields.yml]: This is already defined here: https://github.com/elastic/beats/blob/master/libbeat/processors/add_kubernetes_metadata/_meta/fields.yml#L31

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/test/ksm-unit-1.8 [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# TYPE kube_persistentvolumeclaim_info gauge
+# TYPE kube_persistentvolumeclaim_labels gauge
+# TYPE kube_persistentvolumeclaim_resource_requests_storage_bytes gauge
+# TYPE kube_persistentvolumeclaim_status_phase gauge
+kube_persistentvolumeclaim_info{namespace="default",persistentvolumeclaim="mysql-data",storageclass="rbd",volumename="pvc-mysql-data"} 1
+kube_persistentvolumeclaim_status_phase{namespace="default",persistentvolumeclaim="mysql-data",phase="Bound"} 1
+kube_persistentvolumeclaim_status_phase{namespace="default",persistentvolumeclaim="mysql-data",phase="Lost"} 0
+kube_persistentvolumeclaim_status_phase{namespace="default",persistentvolumeclaim="mysql-data",phase="Pending"} 0
+kube_persistentvolumeclaim_resource_requests_storage_bytes{namespace="default",persistentvolumeclaim="mysql-data"} 1.073741824e+09
+kube_persistentvolumeclaim_labels{label_app="mysql-server",namespace="default",persistentvolumeclaim="mysql-data"} 1
```
Comment (exekias) [inline metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/test/ksm-unit-1.8]: it's interesting how they included labels here when it's normally not part of the info we get. Do you know why?

File: metricbeat/module/kubernetes/_meta/fields.yml (hunk 3 above)
Comment (exekias) [reply metricbeat/module/kubernetes/_meta/fields.yml]: to clarify, that fields.yml is included in any beat using the processor, including Metricbeat

Comment (odacremolbap) [reply metricbeat/module/kubernetes/_meta/fields.yml]: oh, for some reason beyond my comprehension I tested without adding that field and it appeared as non mapped at kibana.

I'll re-check and remove if the field is already there.

thanks!

File: metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/test/ksm-unit-1.8 (hunk 4 above)
Comment (odacremolbap) [reply metricbeat/module/kubernetes/state_persistentvolumeclaim/_meta/test/ksm-unit-1.8]: (not sure if I follow)
that labels pattern seems to be common to all main kube-state-metrics objects. Not seeing it anywhere else, nor for annotations, but it looks like the `label_keyname="value"` is here to stay.

File: metricbeat/module/kubernetes/_meta/fields.yml (hunk 3 above)
Comment (odacremolbap) [reply metricbeat/module/kubernetes/_meta/fields.yml]: done @exekias 
please, re-review if you will

**Comment by odacremolbap** [comment]
jenkins test this

**Comment by odacremolbap** [comment]
jenkins test this

failed tests look unrelated

**Comment by odacremolbap** [comment]
testing notes: https://github.com/odacremolbap/beats/tree/master/metricbeat/module/kubernetes/state_persistentvolumeclaim

**Pull request merged**